chingu-voyage4/Geckos-Team-32
# Merge auth to dev

## Purpose
* Handle EDIT/DELETE routes for user

## Approach/Fixes
* Setup user profile page
* Setup EDIT component for profile page
* Edit username route
  * Handle edge cases
* Delete user entirely
  * Re-route to homepage

## Testing
* Download this PR to your local device
* Run `npm install` to make sure you're up to date on dependencies
* **Ensure your `config` directory is updated as well**
* Startup your local mongoDB with `mongod` in the terminal
* Run `npm run serve` to fire up the app on `localhost://8000`
* Create a new user
* Check to make sure the user has been added in your mongo DB
* Try editing the username
  * New usernames should save
    * Check in the DB that new username has saved
  * If the username is already taken, an alert box should appear
    * We will need to handle these errors differently later
* Try deleting the user account altogether
  * Check in the DB that the user is completely removed 

## Notes
* UI for profile page is minimal (skeleton, no styling)
